patches/server/0880-Allow-usage-of-MiniMessage-in-server-configuration.patch

```diff
@@ -0,0 +1,410 @@
+@@ -1,7 +1,6 @@
+ package com.destroystokyo.paper;
+ 
+ import com.destroystokyo.paper.io.chunk.ChunkTaskManager;
+-import com.google.common.base.Strings;
+ import com.google.common.base.Throwables;
+ 
+ import java.io.File;
+@@ -18,18 +17,23 @@ import java.util.logging.Level;
+ import java.util.regex.Pattern;
+ 
+ import com.google.common.collect.Lists;
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+ import net.minecraft.server.MinecraftServer;
+ import org.bukkit.Bukkit;
+-import org.bukkit.ChatColor;
+ import org.bukkit.command.Command;
+ import org.bukkit.configuration.ConfigurationSection;
+ import org.bukkit.configuration.InvalidConfigurationException;
+ import org.bukkit.configuration.file.YamlConfiguration;
+ import co.aikar.timings.Timings;
+ import co.aikar.timings.TimingsManager;
++import org.jetbrains.annotations.Nullable;
+ import org.spigotmc.SpigotConfig;
+ import org.spigotmc.WatchdogThread;
+ 
++import static net.kyori.adventure.text.minimessage.MiniMessage.miniMessage;
++import static net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.*;
++
+ public class PaperConfig {
+ 
+     private static File CONFIG_FILE;
+@@ -71,8 +75,8 @@ public class PaperConfig {
+         commands.put("paper", new PaperCommand("paper"));
+         commands.put("mspt", new MSPTCommand("mspt"));
+ 
+-        version = getInt("config-version", 25);
+-        set("config-version", 25);
++        version = getInt("config-version", 26);
++        set("config-version", 26);
+         readConfig(PaperConfig.class, null);
+     }
+ 
+@@ -307,11 +311,20 @@ public class PaperConfig {
+         packetInSpamThreshold = getInt("settings.incoming-packet-spam-threshold", 300);
+     }
+ 
+-    public static String flyingKickPlayerMessage = "Flying is not enabled on this server";
+-    public static String flyingKickVehicleMessage = "Flying is not enabled on this server";
++    public static Component flyingKickPlayerMessage;
++    public static Component flyingKickVehicleMessage;
+     private static void flyingKickMessages() {
+-        flyingKickPlayerMessage = getString("messages.kick.flying-player", flyingKickPlayerMessage);
+-        flyingKickVehicleMessage = getString("messages.kick.flying-vehicle", flyingKickVehicleMessage);
++        String kickPlayer = getString("messages.kick.flying-player", "Flying is not enabled on this server");
++        String kickVehicle = getString("messages.kick.flying-vehicle", "Flying is not enabled on this server");
++
++        if (configurationUsesMiniMessage) {
++            flyingKickPlayerMessage = miniMessage().deserialize(kickPlayer);
++            flyingKickVehicleMessage = miniMessage().deserialize(kickVehicle);
++
++        } else {
++            flyingKickPlayerMessage = legacyAmpersand().deserialize(kickPlayer);
++            flyingKickVehicleMessage = legacyAmpersand().deserialize(kickVehicle);
++        }
```

Comment:
Should be done in the patch that adds this.

Reply:
Moved.